docs/content/configuration.md

```diff
@@ -371,19 +371,28 @@ OPA will authenticate using a bearer token obtained through the OAuth2 [client c
 Following successful authentication at the token endpoint the returned token will be cached for subsequent requests for the duration of its lifetime. Note that as per the [OAuth2 standard](https://tools.ietf.org/html/rfc6749#section-2.3.1), only the HTTPS scheme is supported for the token endpoint URL.
 
-| Field | Type | Required | Description |
-| --- | --- | --- | --- |
-| `services[_].credentials.oauth2.token_url` | `string` | Yes | URL pointing to the token endpoint at the OAuth2 authorization server. |
-| `services[_].credentials.oauth2.grant_type` | `string` | No | Defaults to `client_credentials`. |
-| `services[_].credentials.oauth2.client_id` | `string` | No | The client ID to use for authentication. |
-| `services[_].credentials.oauth2.signing_key` | `string` | Yes | Reference to private key used for signing the JWT. |
-| `services[_].credentials.oauth2.thumbprint` | `string` | No | Certificate thumbprint to use for x5t header generation. |
-| `services[_].credentials.oauth2.additional_claims` | `map` | No | Map of claims to include in the JWT (see notes below) |
-| `services[_].credentials.oauth2.include_jti_claim` | `bool` | No | Include a uniquely generated `jti` claim in any issued JWT |
-| `services[_].credentials.oauth2.scopes` | `[]string` | No | Optional list of scopes to request for the token. |
+| Field                                              | Type       | Required | Description                                                                                                                                                                                     |
+|----------------------------------------------------|------------|----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `services[_].credentials.oauth2.token_url`         | `string`   | Yes      | URL pointing to the token endpoint at the OAuth2 authorization server.                                                                                                                          |
+| `services[_].credentials.oauth2.grant_type`        | `string`   | No       | Defaults to `client_credentials`.                                                                                                                                                               |
+| `services[_].credentials.oauth2.client_id`         | `string`   | No       | The client ID to use for authentication.                                                                                                                                                        |
+| `services[_].credentials.oauth2.signing_key`       | `string`   | No       | Reference to private key used for signing the JWT. Required if `aws_kms` is not provided                                                                                                        |
```

Comment:
**Required if `aws_kms` is not provided** this means if someone does not specify `aws_kms`, then this field has to be set? If that's the case then this would be backward incompatible. Or is this the current behavior already?

Reply:
It was a "Required" field before, with this feature you can either pass a private key or a KMS key.

Reply:
Ok. Thanks for the clarification.